### DISCUSSION TITLE
How to pull changes form here without losing my tweaks to the system?

### DISCUSSION THREAD

**Discussion by Ramy-Ashraf** [prasanthrangan/hyprdots#531] [Q&A] [answered]
Every time I pull changes, I lose my changes and my vscode extensions. So, I am asking this question to know the proper way to get what's new without losing everything. (I don't really understand how restore script work)

**Comment by kRHYME7** [comment, marked as answer]
Hello @Ramy-Ashraf,

The recent commit should not affect your VSCode extensions, as we have implemented a feature that allows you to install the repository without overwriting your local configurations.

The `restore` script is designed to pull changes from the repository and automatically back up your latest configuration to `~/.config/cfg*(Date and time)`. It then copies the latest update to the `~/.configs` folder, which means it backs up the latest repository and overrides your `~/.config`.

You have control over which configurations should be overwritten using the control file located at `~/Hyprdots/Scripts/restore_cfg.lst`. The first column indicates whether you want to overwrite the configuration (Y for Yes, N for No). The second column should be the destination directory or file, and the last column is a check if you have that particular application that uses this config.

Here is an example of how the `restore_cfg.lst` file might look:

```bash
❯ cat ~Hyprdots/Scripts/restore_cfg.lst
N|${HOME}|.zshrc .p10k.zsh|zsh oh-my-zsh-git zsh-theme-powerlevel10k-git pokemon-colorscripts-git
Y|${HOME}/.config/Code/User|settings.json|visual-studio-code-bin
Y|${HOME}/.config|dunst|dunst
Y|${HOME}/.config/hypr|animations.conf hyprland.conf keybindings.conf theme.ctl windowrules.conf scripts themes wallbash|hyprland
Y|${HOME}/.config|kitty|kitty
Y|${HOME}/.config|Kvantum|kvantum
Y|${HOME}/.config|MangoHud|mangohud
Y|${HOME}/.config|neofetch|neofetch
Y|${HOME}/.config|qt5ct|qt5ct
Y|${HOME}/.config|rofi|rofi
Y|${HOME}/.config|swaylock|swaylock-effects
N|${HOME}/.config|swww|swww
Y|${HOME}/.config|waybar|waybar
Y|${HOME}/.config|wlogout|wlogout
Y|${HOME}/.config|nwg-look gtk-3.0 xsettingsd|nwg-look
Y|${HOME}/.icons|default|nwg-look
Y|${HOME}|.gtkrc-2.0|nwg-look
Y|${HOME}/.config|dolphinrc kdeglobals|dolphin
Y|${HOME}/.local/share|dolphin|dolphin
Y|${HOME}/.local/share/kxmlgui5|dolphin|dolphin
Y|${HOME}/.local/bin|Hyprdots|hyprland
```

If you have your specific configurations, you may change the Y flag to N. Or you can check your `~/.config/cfg*(latest timedate)` and check for changes or whatever you want to do.

If you want to pull something from the repository, I would suggest creating a fork of this repository and on that fork apply/modify your own preferences. Every time the Hyprdots repository updates, you can merge those changes to your repository. However, learning the basics of Git is a prerequisite. As of now, using Git is the best way to ensure your changes are saved [Source 3](https://www.delftstack.com/howto/git/git-pull-ignore-local-changes/), [Source 6](https://www.atlassian.com/git/tutorials/undoing-changes), [Source 7](https://happygitwithr.com/pull-tricky.html), [Source 8](https://www.freecodecamp.org/news/git-pull-force-how-to-overwrite-local-changes-with-git/), [Source 9](https://github.com/git-guides/git-pull).

**Comment by kRHYME7** [comment]
Also if your wallpapers where already cached you I can flag N on the ``` N|${HOME}/.config|swww|swww ```
line so that on your updates it would be faster

**Comment by Ramy-Ashraf** [comment]
Thanks for your answer. So if I want to update, I can use the restore script not install script. Right?

**Reply by kRHYME7** [reply]
To restore your configurations, you should use the ./install -r script. This script is specifically designed to restore your configurations. If you feel that something is missing, you can try to use ./install. This script installs the packages in case some updates are required as dependencies. However, ./install -r should be enough in most cases.

For more detailed instructions, you can check out the [Installation Wiki](https://github.com/prasanthrangan/hyprdots/wiki/Installation). It provides a comprehensive guide on how to install and restore configurations.

If you have any further questions or need assistance, feel free to ask here in the discussion panel. We're happy to help!

**Comment by kRHYME7** [comment]
Hello, I managed to get a  functional script to sync your keybindings also the hyprland.conf. Would you like to test it?  Not yet complete though.

**Reply by Ramy-Ashraf** [reply]
Sorry for replying late. I only changed 1 keybinding and set the scale to 1 for my screen. So, I don't need the script now. But I am looking forward to see your script and try it.
Thanks for your work.

**Reply by kRHYME7** [reply]
No worries this work is pending for as I discovered a different approach. Thanks for the reply !
